src/Math-Core/Number.extension.st

```diff
@@ -11,3 +11,9 @@ Number >> addWithVector: aVector [
 	"Adds itself to each element of the vector"
 	^ aVector collect: [ :each | each + self ]
 ]
+
+{ #category : #'*Math-Core' }
+Number >> productWithVector: aVector [
```

Comment:
This represents a scalar multiplied by a vector?

Perhaps `timesVector: aVector`